provider/uprops/src/script.rs

```diff
@@ -173,5 +173,157 @@ mod tests {
+        assert!(scx.has_script(0x30A0, Script::Katakana));
+        assert!(!scx.has_script(0x30A0, Script::Common));
+        assert!(!scx.has_script(0x30A0, Script::Inherited));
+
+        // U+0964 DEVANAGARI DANDA
+        assert!(!scx.has_script(0x0964, Script::Common));
+        assert!(scx.has_script(0x0964, Script::Devanagari));
+        assert!(scx.has_script(0x0964, Script::Bengali));
+
+        // TestHasScript() test cases from ICU4J
+
+        // U+063F ARABIC LETTER FARSI YEH WITH THREE DOTS ABOVE
+        assert!(!scx.has_script(0x063F, Script::Common));
+        assert!(scx.has_script(0x063F, Script::Arabic)); // main Script value
+        assert!(!scx.has_script(0x063F, Script::Syriac));
+        assert!(!scx.has_script(0x063F, Script::Thaana));
+
+        // U+0640 ARABIC TATWEEL
+        assert!(!scx.has_script(0x0640, Script::Common)); // main Script value
+        assert!(scx.has_script(0x0640, Script::Arabic));
+        assert!(scx.has_script(0x0640, Script::Syriac));
+        assert!(!scx.has_script(0x0640, Script::Thaana));
+
+        // U+0650 ARABIC KASRA
+        assert!(!scx.has_script(0x0650, Script::Inherited)); // main Script value
+        assert!(scx.has_script(0x0650, Script::Arabic));
+        assert!(scx.has_script(0x0650, Script::Syriac));
+        assert!(!scx.has_script(0x0650, Script::Thaana));
+
+        // U+0660 ARABIC-INDIC DIGIT ZERO
+        assert!(!scx.has_script(0x0660, Script::Common)); // main Script value
+        assert!(scx.has_script(0x0660, Script::Arabic));
+        assert!(!scx.has_script(0x0660, Script::Syriac));
+        assert!(scx.has_script(0x0660, Script::Thaana));
+
+        // U+FDF2 ARABIC LIGATURE ALLAH ISOLATED FORM
+        assert!(!scx.has_script(0xFDF2, Script::Common));
+        assert!(scx.has_script(0xFDF2, Script::Arabic)); // main Script value
+        assert!(!scx.has_script(0xFDF2, Script::Syriac));
+        assert!(scx.has_script(0xFDF2, Script::Thaana));
+
+        // The ICU4J comment for this test says:
+        // An unguarded implementation might go into an infinite loop.
+        assert!(!scx.has_script(0x0640, Script(0xAFFE)));
+    }
+
+    #[test]
+    fn test_script_ext_ranges() {
+        let root_dir = icu_testdata::paths::uprops_toml_root();
+        let provider = ScriptExtensionsPropertyProvider::try_new(&root_dir)
+            .expect("TOML should load successfully");
+
+        let payload: DataPayload<ScriptExtensionsPropertyV1Marker> = provider
+            .load_payload(&DataRequest {
+                resource_path: ResourcePath {
+                    key: key::SCRIPT_EXTENSIONS_V1,
+                    options: ResourceOptions::default(),
+                },
+            })
+            .expect("The data should be valid")
+            .take_payload()
+            .expect("Loading was successful");
+
+        let scx: &ScriptExtensions = &payload.get().data;
+
+        let grantha = scx.get_script_extensions_set(Script::Grantha);
```

Comment:
Nit: Make these tests more thorough, testing both positive and negative, as well as testing edge cases like what happens around the boundary of code points that are included and not included in the SCX set.

The reason I ask is because this test indirectly tests the big `get_range` function, which is inadequately tested due to the limited CPT test data we have in that crate.  We have all the test data over here, and we should use it to more thoroughly exercise that code.